crates/bevy_ecs/src/system/mod.rs

```diff
@@ -96,6 +96,38 @@ pub fn assert_is_system<In, Out, Params, S: IntoSystem<In, Out, Params>>(sys: S)
     }
 }
 
+/// Ensure that a given function is an exclusive system
+///
+/// This should be used when writing doc examples,
+/// to confirm that systems used in an example are
+/// valid exclusive systems
+///
+/// Passing assert
+/// ```
+/// # use bevy_ecs::prelude::World;
+/// # use bevy_ecs::system::assert_is_exclusive_system;
+/// fn an_exclusive_system(_world: &mut World) {}
+///
+/// assert_is_exclusive_system(an_exclusive_system);
+/// ```
+///
+/// Failing assert
+/// ```compile_fail
+/// # use bevy_ecs::prelude::World;
+/// # use bevy_ecs::system::assert_is_exclusive_system;
+/// fn not_an_exclusive_system(_world: &mut World, number: f32) {}
+///
+/// assert_is_exclusive_system(not_an_exclusive_system);
+/// ```
+pub fn assert_is_exclusive_system<
+    Params,
+    SystemType,
+    S: IntoExclusiveSystem<Params, SystemType>,
+>(
+    _sys: S,
```

Comment:
```suggestion
>(
    _sys: impl IntoExclusiveSystem<Params, SystemType>,
```
This is preferred when using a generic for a parameter only. Should make rustfmt happier too :)